### DISCUSSION TITLE
Notify server that push delivered

### DISCUSSION THREAD

**Discussion by Sergey-lang** [invertase/react-native-firebase#6781] [Q&A]
Hi.
Is there any way to notify your server that the message has been delivered to the device? It doesn't matter, closed, open. The main thing is just to find out that it has arrived, so that you can then send undelivered messages through the telecom operator.

**Comment by mikehardy** [comment]
Yes? But maybe not reliably? I cannot answer without more details about what exactly you are sending as message contents (specifying the exact JSON you send as an FCM as well as your exact use case would have helped: https://stackoverflow.com/help/how-to-ask) but in general you should rephrase your question as this, and then read the relevant documentation from firebase about message types and message delivery and here + react-native-firebase about what handlers are called and how to set them up:

What handlers are called, and when, in my application when I send different types of JSON as an FCM?

Then you code your handlers to contact your server as needed.

**Reply by Sergey-lang** [reply]
@mikehardy 
For example, I want to notify my server that a message has been delivered. It doesn't matter how and what to send. Good. For example, I will send the device token to the server when the push message has been delivered. I will know that this user has received the message.
We are talking about NOTIFICATIONS with the mandatory "notification" field.
What handlers, what are we talking about? onMessage and backgrounMessageListener. I didn't see another one in the documentation. Adding queries to these methods breaks the display - notifications are not displayed at all.

**Reply by mikehardy** [reply]
If you send FCM with JSON that contains a notification block, your code will not be executed until the user interacts with the notification on Android (if they do) and iOS with the *option* on iOS of having a notification extension helper installed in iOS at which point you have a limited time to execute code prior to display of the notification. So this requirement (pingback to server on delivery) may be hard or impossible to meet based on the information you provided.

If you split FCM types and use data-only on Android they *should* be delivered reliably and allow handler execution, and with mixed or notification FCM on iOS they *should* be delivered reliably allowing notification extension execution

**Reply by Sergey-lang** [reply]
@mikehardy 
I tested it. Notifications without a notification block do not cause any reaction on the phone at all. That is, if the phone is simply turned off and the application does not work in the background, then nothing happens. Accordingly, there is no point in rejecting notifications that do not come and do not allow the user to pay attention to themselves. Apple's documentation states that content-mutable does not guarantee message delivery. I've been checking different options.
Maybe I don't understand something, but the so-called data messages only work if the application is in the background.

**Reply by mikehardy** [reply]
Data messages on android can actually boot the app if I understand correctly. On iOS they are unreliable so not worth discussing for your use case

**Comment by hongjumeow** [comment]
Dooray! 메일 발송 실패 안내



  
  
    
    
  
  
    
메일 발송
실패 안내
    
  
  
    
        
          
***@***.***) 
          님께
보낸
메일이
전송되지
못하였습니다.
        
        
          실패 사유를 확인해보세요.
        
    
  
  
    
      
        
          
            * 받는 사람 : 
***@***.***) 
          
          
            * 발송 시간 : 
2022-12-31T23:23:26
          
          
            * 메일 제목 : 
Re: [invertase/react-native-firebase] Notify server that push delivered (Discussion #6781)
          
          
            
              
                * 실패 사유 : 
              
              
                
받는 사람이 회원님의 메일을 수신차단 하였습니다.
                
              
            
                      
          
        
      
    
  
  
    
        
          이 메일은 발신전용으로 회신되지 않습니다.
          더 궁금하신 사항은
          ***@***.***
          으로 문의해 주시기 바랍니다.
        
    
  
  
    
        © Dooray!.

**Comment by ChromeUniverse** [comment]
Apologies in advance for an off-topic comment @mikehardy, but I've been receiving emails of this Q&A thread for the past  day because I've been supposedly mentioned here, even though I've never heard of this repo before. I believe @hongjumeow has something to do with this - I've seen countless other Issues and Q&A threads getting hijacked by bots with NSFW links and spam content this past week, but I don't know what can be done about it. Just wanted to give y'all a heads up. Happy New Year's Eve.

**Reply by mikehardy** [reply]
Yes, it appears that this user subscribed for some reason, and has an auto-responder set so every time someone comments here they auto-reply (which posts a comment). Bizarre behavior. Luckily not spam though, that does happen often enough and I work through it.

Personally, you may just unsubscribe of course and leave this whole mess out of your inbox ;-). Happy New Year

**Comment by hongjumeow** [comment]
Dooray! 메일 발송 실패 안내



  
  
    
    
  
  
    
메일 발송
실패 안내
    
  
  
    
        
          
***@***.***) 
          님께
보낸
메일이
전송되지
못하였습니다.
        
        
          실패 사유를 확인해보세요.
        
    
  
  
    
      
        
          
            * 받는 사람 : 
***@***.***) 
          
          
            * 발송 시간 : 
2022-12-31T23:31:36
          
          
            * 메일 제목 : 
Re: [invertase/react-native-firebase] Notify server that push delivered (Discussion #6781)
          
          
            
              
                * 실패 사유 : 
              
              
                
받는 사람이 회원님의 메일을 수신차단 하였습니다.
                
              
            
                      
          
        
      
    
  
  
    
        
          이 메일은 발신전용으로 회신되지 않습니다.
          더 궁금하신 사항은
          ***@***.***
          으로 문의해 주시기 바랍니다.
        
    
  
  
    
        © Dooray!.

**Comment by hongjumeow** [comment]
Dooray! 메일 발송 실패 안내



  
  
    
    
  
  
    
메일 발송
실패 안내
    
  
  
    
        
          
***@***.***) 
          님께
보낸
메일이
전송되지
못하였습니다.
        
        
          실패 사유를 확인해보세요.
        
    
  
  
    
      
        
          
            * 받는 사람 : 
***@***.***) 
          
          
            * 발송 시간 : 
2022-12-31T23:42:21
          
          
            * 메일 제목 : 
Re: [invertase/react-native-firebase] Notify server that push delivered (Discussion #6781)
          
          
            
              
                * 실패 사유 : 
              
              
                
받는 사람이 회원님의 메일을 수신차단 하였습니다.
                
              
            
                      
          
        
      
    
  
  
    
        
          이 메일은 발신전용으로 회신되지 않습니다.
          더 궁금하신 사항은
          ***@***.***
          으로 문의해 주시기 바랍니다.
        
    
  
  
    
        © Dooray!.

**Comment by hongjumeow** [comment]
Dooray! 메일 발송 실패 안내



  
  
    
    
  
  
    
메일 발송
실패 안내
    
  
  
    
        
          
***@***.***) 
          님께
보낸
메일이
전송되지
못하였습니다.
        
        
          실패 사유를 확인해보세요.
        
    
  
  
    
      
        
          
            * 받는 사람 : 
***@***.***) 
          
          
            * 발송 시간 : 
2022-12-31T23:43:54
          
          
            * 메일 제목 : 
Re: [invertase/react-native-firebase] Notify server that push delivered (Discussion #6781)
          
          
            
              
                * 실패 사유 : 
              
              
                
받는 사람이 회원님의 메일을 수신차단 하였습니다.
                
              
            
                      
          
        
      
    
  
  
    
        
          이 메일은 발신전용으로 회신되지 않습니다.
          더 궁금하신 사항은
          ***@***.***
          으로 문의해 주시기 바랍니다.
        
    
  
  
    
        © Dooray!.
